src/esm_runscripts/prepare.py

```diff
@@ -733,7 +734,28 @@ def set_logfile(config):
 
 
 def check_config_for_warnings_errors(config):
+    """
+    Wrapper method to call ``warn_error`` from the `ESM-Tools` recipes. Loops through
+    the trigger variables to call ``warn_error``, only including ``warning`` in the
+    ``triggers`` if it is an interactive session (i.e. user just submitted a job), or
+    if it's not a resubmission (to avoid displaying the warning twice).
+
+    .. note:: use this in the recipe after the ``choose_`` blocks are resolved,
```

Comment:
I have a silly idea....can we check for this? The `recipe` in in `config` after all. I will make a separate issue with some ideas about sanity checks for recipe order.